runtime/major_gc.c

```diff
@@ -757,7 +759,8 @@ void caml_darken_cont(value cont)
               With_status_hd(hd, NOT_MARKABLE))) {
         value stk = Field(cont, 0);
         if (Ptr_val(stk) != NULL)
-          caml_scan_stack(&caml_darken, 0, Ptr_val(stk), 0);
+          caml_scan_stack(&caml_darken, darken_scanning_flags, 0,
```

Comment:
```suggestion
          caml_scan_stack(&caml_darken, SCANNING_GENERIC, 0,
```
etc. in other places, in accordance with the suggestion of having a zero field.

Reply:
I thought again about having a zero field, but I don't like it. (For once, I don't know of a good name for the field; `SCANNING_FLAGS_NONE` is the least meh but still not exciting.)

I went to look at how other flags-enum do this in the codebase, and it looks like some have an explicit name for the empty set, others don't. (For example the `CHANNEL` flags in caml/io.h don't have one, or the marshalling flags in `extern.c`, but the `SELECT_MODE` flags in unix/select_win32.c do have one.)

Reply:
If your flags are a set (of logically-ORed powers of two), then `0` is perfect because it clearly denotes the empty set.  (Squint a bit and you'll see ∅.  Or remember your set theory class, the part where natural numbers are constructed.)